src/java/org/apache/cassandra/db/ReadExecutionController.java

```diff
@@ -68,6 +73,8 @@ public class ReadExecutionController implements AutoCloseable
         this.command = command;
         this.createdAtNanos = createdAtNanos;
 
+        this.sstablesScannedPerRowRead = new ClearableHistogram(new DecayingEstimatedHistogramReservoir(true));
```

Comment:
This is initialized per read query, right? Is it worth only initializing when `Tracing.traceSinglePartitions()` is true? It seems like it might be an additional optimization to only track when we're going to retrieve it.

Reply:
Also, what is the motivation for selecting `new ClearableHistogram(new DecayingEstimatedHistogramReservoir(true));` instead of some other histogram?

Reply:
@michaeljmarshall  As I understand, Tracing can be enabled dynamically. So I decided to collect data always to avoid null checks and such + to get output after the long query if the tracing got enabled mid-query.
It should not be too heavy, it is used in metrics/codahale.

I'll switch to regular Histogram sine clear() is not used.